MantidPlot/src/ApplicationWindow.cpp

```diff
@@ -14041,9 +14042,30 @@ void ApplicationWindow::updateRecentFilesList(QString fname) {
     recentFiles.pop_back();
 
   recentFilesMenu->clear();
-  for (int i = 0; i < (int)recentFiles.size(); i++)
-    recentFilesMenu->insertItem("&" + QString::number(i + 1) + " " +
-                                recentFiles[i]);
+  for (int i = 0; i < (int)recentFiles.size(); i++) {
+    QMenuItem *mi = new QMenuItem;
+
+    try {
+      Mantid::API::MultipleFileProperty mfp("tester");
+      mfp.setValue(recentFiles[i].toStdString());
+      const std::vector<std::string> files =
+          Mantid::API::MultipleFileProperty::flattenFileNames(mfp());
+      if (files.size() == 1) {
+        mi->setText("&" + QString::number(i + 1) + " " + recentFiles[i]);
+      } else {
+        std::ostringstream ostr;
+        ostr << "&" << QString::number(i + 1).toStdString() << " " << files[0]
```

Comment:
This can crash with a SegFault, bypassing the catch, because `mfp.setValue` strips out any filenames that cannot be resolved. So if your recent file history contains a file that you have since deleted or renamed, `files` will be empty and `files[0]` goes boom. The crash occurs immediately on startup, before the main window shows up.